src/binary-reader-ir.cc

```diff
@@ -660,6 +720,7 @@ Result BinaryReaderIR::OnFunctionBodyCount(Index count) {
 
 Result BinaryReaderIR::BeginFunctionBody(Index index, Offset size) {
   current_func_ = module_->funcs[index];
+  current_func_->loc = GetLocation();
```

Comment:
Seems odd that this was not needed before.. do you know why this is needed now?

Reply:
I am not sure why it wasn't used before, and maybe I am using it as a hack?

Anyway, It is now needed by CodeMetadataExprQueue: the offsets are always absolute but code metadata need relative offsets from the beginning of the function. So it seemed natural to fill this existing field here and use it to get relative offsets later

Reply:
Oh, actually you are adding this new `loc` field to `struct Func`.   I guess that makes sense for this use case and we already track this for other things such as expressions, and I guess the location a function is different from the location of its first expression due to param declaration?

Reply:
Sorry, I made this change some time ago and I didn't remember that I added the field!

But yes, it felt natural, since like you say it is there for similar things.

And also yes, the difference with the location of the first expression is that there are the locals in between.